Signal/src/call/CallService.swift

```diff
@@ -1152,6 +1161,33 @@ protocol CallServiceObserver: class {
         // Do nothing
     }
 
+    internal func callRecordDidChange(call: SignalCall, callRecord: TSCall?) {
+        AssertIsOnMainThread()
+
+        updateCallRecordType(call: call)
+    }
+
+    internal func updateCallRecordType(call: SignalCall?) {
```

Comment:
I'd love to decompose CallService, but I'm ambivalent about this. It moves business logic into the model.  I prefer dumb models.  Anyway, I've done this.